File: backend/src/routes/weather.ts
import { Router, Request, Response } from 'express';
import { getWeatherAndRecommendation } from '../services/weatherService.js';

const router = Router();

router.get('/current', (req: Request, res: Response) => {
  const lat = req.query.lat ? parseFloat(req.query.lat as string) : -7.25;
  const lon = req.query.lon ? parseFloat(req.query.lon as string) : 112.75;
  const village = (req.query.village as string) || "Desa Sukamaju";

  const data = getWeatherAndRecommendation(lat, lon, village);
  res.json(data);
});

export default router;

File: backend/legacy_python/app/api/weather.py
from fastapi import APIRouter, Query
from app.models.schemas import WeatherRecommendation
from app.services.weather_service import get_weather_and_recommendation

router = APIRouter(prefix="/weather", tags=["Cuaca & Rekomendasi Lahan"])

@router.get("/current", response_model=WeatherRecommendation)
def get_current_weather(
    lat: float = Query(-7.25, description="Latitude lokasi lahan"),
    lon: float = Query(112.75, description="Longitude lokasi lahan"),
    village: str = Query("Desa Sukamaju", description="Nama desa/kelurahan")
):
    return get_weather_and_recommendation(lat=lat, lon=lon, village=village)
